src/tqec/_cli/subcommands/run_example.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import annotations
+
+
+import argparse
+import logging
+from pathlib import Path
+
+import numpy as np
+import matplotlib.pyplot as plt
+import sinter
+from typing_extensions import override
+
+from tqec._cli.subcommands.base import TQECSubCommand
+from tqec.compile.specs.library.css import CSS_BLOCK_BUILDER, CSS_SUBSTITUTION_BUILDER
+from tqec.compile.specs.library.zxxz import (
+    ZXXZ_BLOCK_BUILDER,
+    ZXXZ_SUBSTITUTION_BUILDER,
+)
+from tqec.gallery import logical_cnot_block_graph
+from tqec.noise_models import NoiseModel
+from tqec.simulation.plotting.inset import plot_observable_as_inset
+from tqec.simulation.simulation import start_simulation_using_sinter
+
+
+class RunExamplesTQECSubCommand(TQECSubCommand):
+    @staticmethod
+    @override
+    def add_subcommand(
+        main_parser: argparse._SubParsersAction[argparse.ArgumentParser],
+    ) -> None:
+        parser: argparse.ArgumentParser = main_parser.add_parser(
+            "run-examples",
+            description=(
+                "Runs the full pipeline on the CNot example."
+                "block graph -> stim circuits and observables -> simulation -> plots"
+            ),
+        )
+        parser.add_argument(
+            "--out-dir",
+            help="Directory to save the generated outputs.",
+            type=Path,
+            required=True,
+        )
+        parser.add_argument(
+            "-k",
+            help="The scale factors applied to the circuits.",
+            nargs="+",
+            type=int,
+            default=[1, 2, 3, 4],
+        )
+
+        parser.add_argument(
+            "-p",
+            help="The noise levels applied to the simulation.",
+            nargs="+",
+            type=float,
+            default=list(np.logspace(-4, -1, 10)),
+        )
+
+        parser.add_argument(
+            "--obs-include",
+            help=(
+                "The observable indices to be included in the circuits. "
+                "If not provided, all potential observables will be included."
+            ),
+            nargs="*",
+            type=int,
+        )
+        parser.add_argument(
+            "--add-detectors",
+            help="Whether to add detectors to the circuits.",
+            action="store_true",
+        )
+        parser.add_argument(
+            "--generate-only",
+            help="Whether to run generation only.",
+            action="store_true",
+        )
+        parser.add_argument(
+            "--code-style",
+            help="Use the CSS or ZXXZ code style.",
+            choices=["css", "zxxz"],
+        )
+        parser.add_argument(
+            "--include-z-basis",
+            help="Whether to include the Z basis in the simulation.",
+            action="store_true",
+        )
```

Comment:
Maybe change to `--basis {x,z}`? That would be easier to understand in my opinion.